proto/semantic.proto

```diff
@@ -161,12 +162,24 @@ enum NodeType {
   REFERENCE = 4;
 }
 
+enum SyntaxType {
+  UNKNOWN = 0;
```

Comment:
What is `UNKNOWN` needed for?

Reply:
In protobufs, the [default value](https://developers.google.com/protocol-buffers/docs/proto3) for an enum is the first enum value (which must be 0). This makes sure that we can tell the difference between the default and the field being set to something specific.

Reply:
Oh, that makes sense. I thought that all fields were always 'nullable' in a protobuf, but it looks like that is only true for fields whose values are other messages.